src/source-pcap.c

```diff
@@ -641,3 +705,223 @@ void PcapTranslateIPToDevice(char *pcap_dev, size_t len)
 
     pcap_freealldevs(alldevsp);
 }
+
+#ifdef UNITTESTS
+
+static uint32_t Upper32(uint64_t value)
+{
+    return value >> 32;
+}
+static uint32_t Lower32(uint64_t value)
+{
+    return value; /* slice lower 32bits */
+}
+
+static int UpdatePcapStatsValue64NoChange01(void)
+{
+    /*
+     * No change in counter values.
+     * Last count is within first 32bit range, i.e. same as pcap_stat range.
+     */
+    uint64_t last[] = { 0, 12345, (uint64_t)UINT32_MAX };
+    u_int current[] = { 0, 12345, UINT_MAX };
+    FAIL_IF_NOT(ARRAY_SIZE(last) == ARRAY_SIZE(current));
+
+    for (size_t i = 0; i < ARRAY_SIZE(last); ++i) {
+        FAIL_IF_NOT(last[i] == current[i]);
+
+        UpdatePcapStatsValue64(&last[i], current[i]);
+        FAIL_IF_NOT(last[i] == current[i]);
+    }
+
+    PASS;
+}
+
+static int UpdatePcapStatsValue64NoChange02(void)
+{
+    /*
+     * No change in counter values.
+     * Last count is outside 32bits range.
+     */
+    uint64_t last[] = { (2ull << 32) + 0, (3ull << 32) + 12345,
+        (3ull << 32) + (uint64_t)UINT32_MAX,
+        ((uint64_t)UINT32_MAX << 32) + (uint64_t)UINT32_MAX };
+    u_int current[] = { 0, 12345, UINT_MAX, UINT_MAX };
+    FAIL_IF_NOT(ARRAY_SIZE(last) == ARRAY_SIZE(current));
```

Comment:
Doing separate last/current arrays is kind of messy to read.

An alternative would be:
```
/* Structured test data to make it easier on my eyes */
typedef struct TestData_
{
    uint64_t last; /* internal 64bit counter to drag along */
    u_int current; /* 32bit pcap stat */
} TestData;

static int UpdatePcapStatsValue64NoChange02(void)
{
    /*
     * No change in counter values.
     * Last count is outside 32bits range.
     */
    TestData data[] = {{.last = (2ull << 32) + 0, .current = 0},
            {.last = (3ull << 32) + 12345, .current = 12345},
            {.last = (3ull << 32) + (uint64_t)UINT32_MAX, .current = UINT_MAX},
            {.last = UINT64_MAX, .current = UINT_MAX}};

    for (size_t i = 0; i < ARRAY_SIZE(data); ++i) {
        uint32_t upper = Upper32(data[i].last);
        FAIL_IF_NOT(Lower32(data[i].last) == data[i].current);

        UpdatePcapStatsValue64(&data[i].last, data[i].current);
        FAIL_IF_NOT(Lower32(data[i].last) == data[i].current);
        FAIL_IF_NOT(Upper32(data[i].last) == upper);
    }

    PASS;
}

```